core/primitives-core/src/profile.rs

```diff
@@ -19,46 +19,92 @@ impl<T> Clone for FixedArray<T> {
 }
 
 /// Profile of gas consumption.
-pub type ProfileData = FixedArray<u64>;
+#[derive(Clone)]
+pub struct ProfileData {
+    repr: Repr,
+}
 
-pub fn clone_profile(profile: &ProfileData) -> ProfileData {
-    profile.clone()
+#[derive(Clone)]
+enum Repr {
+    #[cfg(feature = "costs_counting")]
+    Enabled {
+        data: FixedArray<u64>,
+    },
+    Disabled,
 }
 
-impl ProfileData {
-    pub fn new() -> Self {
-        Self { data: Rc::new(RefCell::new([0u64; 1 + ExtCosts::count() + ActionCosts::count()])) }
+#[cfg(not(feature = "costs_counting"))]
+const _ASSERT_NO_OP_IF_COUNTING_DISABLED: [(); 0] = [(); std::mem::size_of::<ProfileData>()];
+
+impl Default for ProfileData {
+    fn default() -> ProfileData {
+        ProfileData::new_disabled()
     }
+}
 
+impl ProfileData {
     const EXT_START: usize = 1;
     const ACTION_START: usize = ProfileData::EXT_START + ExtCosts::count();
     // const LENGTH: usize = PROFILE_DATA_LEN;
```

Comment:
should this be removed?

Reply:
Maybe? Haven't really looked into the actual impl here)